### ISSUE TITLE
k6 shows default with red cross during simple test instead of green checkmark

### ISSUE THREAD

**Issue by hecnavsanz** [grafana/k6#2500]
### Brief summary

Hi *,
I don't know why but I get different result from MacOS than Linux when running a simple test `k6 run test.js` with `test.js`:
```
import http from 'k6/http';

export default function () {
  http.get('https://test.k6.io');
}
```
This is the output:
- MacOS
**running (00m00.7s), 0/1 VUs, 1 complete and 0 interrupted iterations
default ✓ [======================================] 1 VUs  00m00.7s/10m0s  1/1 iters, 1 per VU**
- Linux
**running (00m00.5s), 0/1 VUs, 1 complete and 0 interrupted iterations
default ✗ [======================================] 1 VUs  00m00.5s/10m0s  1/1 iters, 1 per VU**

all seems to be fine with --verbose, but in Linux shows the red cross ✗.

MacOS 12.3.1 and `brew install k6`. (green checkmark)
Linux: Oracle/Red Hat/Rocky Linux and `dnf install https://dl.k6.io/rpm/repo.rpm` + `dnf install k6`. (red cross)

The fact is that with a more simple test:
```
let counter = 0;

export default function() {
    counter;
};
```

both MacOS and Linux doesn't show any cross.

Thnx in advance.

### k6 version

k6 v0.37.0 ((devel), go1.18.1, linux/amd64)

### OS

Linux Oracle/Red Hat/Rocky 8

### Steps to reproduce the problem

k6 run test.js

- test.js
import http from 'k6/http';

export default function () {
  http.get('https://test.k6.io');
}

### Expected behaviour

show green checkmark in the default label.

### Actual behaviour

show red cross in the default label.

**Comment by hecnavsanz** [comment]
sorry my fault, I used a more simple example:

let counter = 0;

export default function() {
    counter;
};

well and now it doesn't show red or green cross but seems to be working.

**Comment by hecnavsanz** [comment]
with a simple example it doesn't show anything (MacOS and Linux) ... but at least I don't see the red cross.
thnx.

**Comment by MStoykov** [comment]
Thanks for reporting this @hecnavsanz :bow: 

It seems to be a race condition in the code that updates the UI, which has no effect on anything else. It also only happens for me using GOMAXPROCS=1 - so having only 1 thread that runs go code, which is extremely unlikely in real life. 

I would expect that it was possible even without that ... but much less likely, as I don't remember ever seeing it, and couldn't reproduce it once.

I have a fix for it [here](https://github.com/grafana/k6/pull/2502

Also, it seems to me that you compiled k6 locally as both you are using go version that we haven't built with `1.18.1`. Also, prior to some changes *after* v0.37.0 it seems like instead of a cross, there was nothing-  no tick or cross.  I am mostly mentioning this, so you look into your setup.

**Comment by hecnavsanz** [comment]
Hi Mihail,
thank u for ur help .. I was removing go latest version and using default:

# k6 version
k6 v0.37.0 (2022-03-15T09:43:55+0000/67657f4, go1.17.7, linux/amd64)
#

and as u said nothing shows …

# k6 run test.js
…
running (00m00.0s), 0/1 VUs, 1 complete and 0 interrupted iterations
default   [======================================] 1 VUs  00m00.0s/10m0s  1/1 iters, 1 per VU


but when I was using --vus  w/ 1 or 2 … now  shows a down arrow:

# k6 run test.js --vus 2 --duration 10s
running (10.0s), 0/1 VUs, 1985407 complete and 0 interrupted iterations
default ↓ [======================================] 1 VUs  10s

I’m not sure it clarifies a bit more. Cheers.


From: Mihail Stoykov ***@***.***>
Date: Thursday, April 21, 2022 at 10:03
To: grafana/k6 ***@***.***>
Cc: hecnavsanz ***@***.***>, Mention ***@***.***>
Subject: Re: [grafana/k6] k6 shows default with red cross during simple test instead of green checkmark (Issue #2500)

Thanks for reporting this @hecnavsanz<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fhecnavsanz&data=05%7C01%7C%7Cb30a31ea36e342ba303f08da23a812d6%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637861502015485050%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=KCFqJeat8uzplDR0%2B3SigSQFw%2Fj1PN7q0NlexWi%2FyDM%3D&reserved=0> 🙇

It seems to be a race condition in the code that updates the UI, which has no effect on anything else. It also only happens for me using GOMAXPROCS=1 - so having only 1 thread that runs go code, which is extremely unlikely in real life.

I would expect that it was possible even without that ... but much less likely, as I don't remember ever seeing it, and couldn't reproduce it once.

I have a fix for it [here](#2502<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fgrafana%2Fk6%2Fpull%2F2502&data=05%7C01%7C%7Cb30a31ea36e342ba303f08da23a812d6%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637861502015485050%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=cINDCJyBDdCQ0qYL4W0AJIMdmF6Frj2%2F%2Fp%2BUHXZj5uE%3D&reserved=0>

Also, it seems to me that you compiled k6 locally as both you are using go version that we haven't built with 1.18.1. Also, prior to some changes after v0.37.0 it seems like instead of a cross, there was nothing- no tick or cross. I am mostly mentioning this, so you look into your setup.
